### PR TITLE
core: replace Obj with standard smart pointer #522

### PR DESCRIPTION
**Pull request by malysh123**
Fixes [#522](https://github.com/epam/Indigo/issues/522)

### CODE REVIEW

File: core/indigo-core/molecule/src/inchi_wrapper.cpp [hunk 1]
```diff
@@ -603,10 +603,11 @@ void InchiWrapper::saveMoleculeIntoInchi(Molecule& mol, Array<char>& inchi)
         }
 
     Molecule* target = &mol;
-    Obj<Molecule> dearom;
+    std::shared_ptr<Molecule> dearom;
+
     if (has_aromatic)
     {
-        dearom.create();
+        dearom = std::shared_ptr<Molecule>();
```
Comment (mkviatkovskii) [inline core/indigo-core/molecule/src/inchi_wrapper.cpp:610]: Looks like we can use `unique_ptr` here, lifetime of the object is straight and limited. Moreover, we can move declaration of `dearom` inside `if` clause.

File: core/indigo-core/molecule/src/molecule_fingerprint.cpp [hunk 2]
```diff
@@ -600,12 +600,12 @@ void MoleculeFingerprintBuilder::_handleSubgraph(Graph& graph, const Array<int>&
...
-    Obj<TautomerSuperStructure> tau_super_structure;
+    std::shared_ptr<TautomerSuperStructure> tau_super_structure;
     BaseMolecule* mol_for_enumeration = &mol;
 
     if (!query && _parameters.tau_qwords > 0 && !skip_tau)
     {
-        tau_super_structure.create(mol.asMolecule());
+        tau_super_structure = std::make_shared<TautomerSuperStructure>(mol.asMolecule());
 
         _tau_super_structure = tau_super_structure.get();
```
Comment (mkviatkovskii) [inline core/indigo-core/molecule/src/molecule_fingerprint.cpp:610]: I believe it's better to use `unique_ptr` here and even change type of `_tau_super_structure` to `std::unique_ptr`.

Comment (malysh123) [reply core/indigo-core/molecule/src/molecule_fingerprint.cpp:610]: But we have assignment operator here.

Comment (mkviatkovskii) [reply core/indigo-core/molecule/src/molecule_fingerprint.cpp]: Let's take a look at this and discuss together.

**Pull request closed without merging**